File: micronaut-mcp-server-java-sdk/src/test/java/io/micronaut/mcp/server/stateless/sync/resources/PgnFile.java
package io.micronaut.mcp.server.stateless.sync.resources;

import io.micronaut.context.annotation.EachProperty;
import io.micronaut.context.annotation.Parameter;
import io.micronaut.context.annotation.Requires;
import io.micronaut.core.naming.Named;

@Requires(property = "spec.name", value = "StatelessSyncResourceListTest")
@EachProperty("pgn")
class PgnFile implements Named {
    private final String name;
    private String path;
    private Integer round;

    PgnFile(@Parameter String name) {
        this.name = name;
    }

    public Integer getRound() {
        return round;
    }

    public void setRound(Integer round) {
        this.round = round;
    }

    @Override
    public String getName() {
        return name;
    }

    public String getPath() {
        return path;
    }

    public void setPath(String path) {
        this.path = path;
    }
}

File: test-suite-docs-python/src/test/python/micronaut/mcp/docs/resources/PgnFile.py
from typing import Annotated

from micronaut.context.annotation import EachProperty, Parameter, Requires
from micronaut.core.naming import Named


@Requires(property="spec.name", value="ResourcesFactoryTest")
@EachProperty("pgn")
class PgnFile(Named):
    """Configuration of a PGN file: ``pgn.<name>.path`` and ``pgn.<name>.round``."""

    path: str | None = None
    round: int | None = None

    def __init__(self, name: Annotated[str, Parameter]):
        self.name = name
